server/jobs/job_store.go

```diff
@@ -132,13 +139,15 @@ func (j *LayeredJobStore) SetJobCompleteStatus(jobID string, status JobStatus) e
 	job := j.jobs[jobID]
 	job.Status = Complete
 
+	reader, size := job.GetReader()
+
 	// Persist to storage backend
-	ok, err := j.storageBackend.Write(jobID, job.Output)
+	ok, err := j.storageBackend.Write(jobID, reader, size)
```

Comment:
Why not just pass in `job.Output` into the backend storage and make backend deal with turning it into a reader? I think @nishkrishnan meant that job's role should just be return what it has and let the other processes handle type conversion. In this case writer should be able to convert our string blob into a reader.